crates/sui-core/src/unit_tests/mysticeti_manager_tests.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::authority::test_authority_builder::TestAuthorityBuilder;
+use crate::checkpoints::CheckpointServiceNoop;
+use crate::consensus_handler::ConsensusHandlerInitializer;
+use crate::consensus_manager::mysticeti_manager::MysticetiManager;
+use crate::consensus_manager::narwhal_manager::narwhal_manager_tests::checkpoint_service_for_testing;
+use crate::consensus_manager::ConsensusManagerMetrics;
+use crate::consensus_manager::ConsensusManagerTrait;
+use crate::consensus_validator::{SuiTxValidator, SuiTxValidatorMetrics};
+use fastcrypto::traits::KeyPair;
+use mysten_metrics::RegistryService;
+use prometheus::Registry;
+use std::sync::Arc;
+use std::time::Duration;
+use sui_swarm_config::network_config_builder::ConfigBuilder;
+use tokio::time::sleep;
+
+#[tokio::test(flavor = "current_thread", start_paused = true)]
+async fn test_mysticeti_manager() {
+    // GIVEN
+    let configs = ConfigBuilder::new_with_temp_dir()
+        .committee_size(1.try_into().unwrap())
+        .build();
+
+    for _i in 0..3 {
```

Comment:
This test is trying to start/shutdown the same validator multiple times. It currently fails with error:
```
---- consensus_manager::mysticeti_manager::mysticeti_manager_tests::test_mysticeti_manager stdout ----
thread 'consensus_manager::mysticeti_manager::mysticeti_manager_tests::test_mysticeti_manager' panicked at /Users/akichidis/.cargo/git/checkouts/mysticeti-7f55ed257fcb2008/2d773a9/mysticeti-core/src/network.rs:89:14:
Failed to bind to local socket: Os { code: 48, kind: AddrInUse, message: "Address already in use" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

which dictates that network doesn't shutdown promptly. I haven't investigated at all why that's the case , I'll have a closer look on it tomorrow.

Reply:
Did you try to remove `flavor = "current_thread", start_paused = true` from the test def, maybe it has something to do with the setup?

Reply:
Looking further on Mysticeti it looks like that not all components shutdown properly. I've tried to fix as part of this PR https://github.com/MystenLabs/mysticeti/pull/43  - hopefully after the issues will get resolved.